relayer/src/types.ts

```diff
@@ -1,13 +1,22 @@
-import { U64, U32 } from "@polkadot/types/primitive";
+import { U64 } from "@polkadot/types/primitive";
 import { Hash } from "@polkadot/types/interfaces";
+import Parachain from "./parachain";
 
 export type TxData = {
   feedId: U64;
   block: string;
   metadata: Metadata;
+  chain: string;
```

Comment:
Might be useful to use branded type here instead of just `string`

Reply:
could you pls elaborate on how that will help?

Reply:
https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-branding-and-type-tagging-6cf6e516523d

By having a branded type you will be able to distinguish between regular strings and strings that have a specific meaning to them, such that code that uses them incorrectly doesn't compile. It is kind of sort of like new types in Rust.

Reply:
I saw that, but I am not sure how we can benefit from those. As far as I understand we have to implement individual type per chain, it's ok while we still have <10 chains, but we're going to have more. And we only use chain name for logging, so we can understand what transaction corresponds to which chain. Also, names can change.

Reply:
No, branded types are **types**, which means it will prevent you from using random strings where block hash, block contents or chain name is expected.

This is not a blocker here, but I would strongly consider to think about it.

Reply:
ok, i think i got it. added